pyoneer/rl/targets/reward_ops.py

```diff
@@ -0,0 +1,668 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+
+@tf.function
+def discounted_returns(rewards,
+                       bootstrap_value=None,
+                       discounts=0.99,
+                       weights=1.0,
+                       time_major=False,
+                       back_prop=False,
+                       name=None):
+    """Compute discounted returns.
+
+    Args:
+        rewards: tensor of shape [Batch x Time], [Batch x Time x ...]
+        bootstrap_value: The last value [t+n]. A tensor of shape [],
+            [...], [Batch], [Batch, ...], [Batch x Time], [Batch x Time x ...].
+        discounts: tensor of shape [], [...], [Batch], [Batch, ...],
+            [Batch x Time], [Batch x Time x ...].
+        weights: tensor of shape [], [...], [Batch], [Batch, ...],
+            [Batch x Time], [Batch x Time x ...].
+        time_major: flag if tensors are time_major.
+            Batch and Time are transposed in this doc.
+        back_prop: allow back_prop through the calculation.
```

Comment:
OOC, in what circumstances would we want to backprop through the return calculation?